ph5/utilities/ph5validate.py

```diff
@@ -749,47 +756,47 @@ def check_event_t_completeness(self, event):
         # EVENT CHECKS
         if not event['id_s']:
             error.append("Event id is missing.")
+        else:
+            try:
+                if not (0 <= int(event['id_s']) <= 65535):
+                    error.append("Event ID '%s' not between 0 and 65535."
+                                 % event['id_s'])
+                elif int(event['id_s']) > 32767:
+                    warning.append("Event ID '%s' is more than 32767. "
+                                   "Not compatible with SEGY revision 1."
+                                   % event['id_s'])
+            except ValueError:
+                error.append("Event ID '%s' not a whole "
+                             "number between 0 and 65535." % event['id_s'])
+
         if not event['description_s']:
             warning.append("Event description is missing.")
         # EVENT LOCATION
-        if event['location/coordinate_system_s'] is None:
-            warning.append("No Event location/coordinate_system_s value "
-                           "found.")
-
-        if event['location/projection_s'] is None:
-            warning.append("No Event location/projection_s value "
-                           "found.")
-
-        if event['location/ellipsoid_s'] is None:
-            warning.append("No Event location/ellipsoid_s value "
-                           "found.")
-
-        if event['location/description_s'] is None:
-            warning.append("No Event location/description_s value "
-                           "found.")
-
+        # removed check for coordinate_system_s, projection_s,
+        # projection_s , ellipsoid_s , description_s
+        # because they are not required and most of PIs do not fill them out
```

Comment:
How significantly do these checks improve the data? I don't know if we should remove checks the make the data more complete. It seems better for network operators to provide us with the most complete datasets possible.

Reply:
The fields: location/coordinate_system_s, projection_s, projection_s , ellipsoid_s  are not available when we use noven to turn a csv of events into a kef.  I believe they are blank in all the event tables I've seen.  Would it help for me to check event tables of some PH5s we've completed and sent (but still have on disk) to see if they are all blank, and also see if there are any that don't have those fields at all?

Reply:
I've got to dash back to testing, but it looks like these fields are not in the kef saved in noven.  They are in the kef output by ph5tokef, so it looks like they're added when keftoph5 is run?  Going slightly astray from the PR, this seems to be true for the array tables too.  (For Fairfield array tables, which are created while segdtoph5 is run instead of using noven, the coordinate system and projection are populated, but ellipsoid is not.)

Reply:
How important are the fields "location/coordinate_system_s, projection_s, projection_s , ellipsoid_s" to the scientific validity of the events table?  Its seems like knowing the latitude and longitude of the event would be important, especially if this was deemed as a warning at some point. It seems like Noven should be updated rather than removing validation checks. What ever is done, there should be consistency in the data creation process and in the data validation process. Our goal is to keep the data as internally consistent as possible. It doesn't seem like we want to hide inconsistency in the data that have been previously exposed.

Reply:
I will look into those fields, but it may take me a while with the more urgent issues going on.  I am pretty sure those fields get populated for Fairfield node experiments because the conversion from UTM to lat-long that takes place when the data is loaded and populates the array tables, uses WGS84, which is a common system.

Should we discuss removing the changes to ph5_validate "location/coordinate_system_s, projection_s, projection_s , ellipsoid_s" and "location/X,Y,Z/unit_s" from this PR?

Reply:
@hrotman-pic, yeah this is the right PR to talk about anything associated with ph5_validate and checking PH5 location information. We should discuss the business logic of what location fields we want to check and how important these specific are. This PR is a good place to suggest alternative changes to the format if we decide fields are irrelevant or we are missing fields as well.